### ISSUE TITLE
Nix smp server package

### ISSUE THREAD

**Issue by ehmry** [simplex-chat/simplexmq#731]
Would be nice to have the smp server packaged in a flake here or downstream in Nixpkgs.

**Comment by ppom0** [comment]
Pushing the server to Hackage would make its packaging in Nixpkgs trivial. Is it doable? It seems like [it was done](https://hackage.haskell.org/package/simplexmq) until v1.1.0.

**Comment by epoberezkin** [comment]
alright, we will re-start publishing packages.

**Comment by turion** [comment]
Any update on this?

**Comment by ehmry** [comment]
I've built the SMP server with Nix but its not suitable for a NixOS service. It uses hardcoded file-system paths, the user is expected to initalize it by running it as root, and the configuration isn't documented.

**Comment by giorgiga** [comment]
I took a look at the code (I also want this packaged for nix), and the hard-coded paths (`/etc/opt/simplex`, `/var/opt/simplex-xftp` and the like) seem to have been addressed shortly after you posted this issue (see #817 for smp-server, I didn't look up the commits for the other binaries, but if you look at the various `Main.rs` they all seem to be using environment variables now).

@ehmry Are you still interested in packaging the smp-server (and possibly the other ones) for nixos? We can work together on that.
